File: Data_Structures_and_Algorithms/Python/Sorting/bubble-sort.py
# Bubble sort - compares two adjacent elements and swaps them until they are in the intended order

def bubbleSort(data):
  # loop to access each array element
  for i in range(len(data)):

    # loop to compare array elements
    for j in range(len(data) - i - 1):

      # compare two adjacent elements
      # change > to < to sort in descending order
      if data[j] > data[j+1]:
        temp = data[j]
        data[j] = data[j+1]
        data[j+1] = temp

data = [-2, 45, 0, 11, -9]

bubbleSort(data)

print('Sorted Array in Ascending Order:')
print(data)

File: Data_Structures_and_Algorithms/Java/BubbleSort.java
package Data_Structures_and_Algorithms.Java;

class BubbleSort {
    public static void main(String[] args) {
        int[] nums = {1,5,6,4,8,2};
        for(int i : nums) {
            System.out.print(i);
        }
        System.out.println();
        bubbleSort(nums);
        
        for(int i : nums) {
            System.out.print(i);
        }
        System.out.println();
    }
    public static void bubbleSort(int[] nums) {
        for(int i = 0; i< nums.length; i++) {
            for(int j = 0;j < nums.length-1-i; j++) {
                if(nums[j] > nums[j+1]) {
                    int temp = nums[j];
                    nums[j] = nums[j+1];
                    nums[j+1] = temp; 
                }
            }
        }
    }
}